### PR TITLE
Added Brocade IronWare interface dBm sensor support

### PR DESCRIPTION
**Pull request by thecityofguanyu**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/discovery/sensors/dbm/ironware.inc.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+$multiplier = 1;
+$divisor = 1000;
+
+foreach ($pre_cache['ironware_optic_oids'] as $index => $entry) {
+    if ($entry['snIfOpticalMonitoringRxPower']) {
+        if (strpos($entry['snIfOpticalMonitoringRxPower'], 'N\/A') !== true) {
+            $oid = '.1.3.6.1.4.1.1991.1.1.3.3.6.1.3.' . $index;
+            $descr = dbFetchCell('SELECT `ifDescr` FROM `ports` WHERE `ifIndex`= ? AND `device_id` = ?', array($index, $device['device_id'])) . ' Rx Power';
+            $currentrx = $entry['snIfOpticalMonitoringRxPower'];
```
Comment (laf) [inline includes/discovery/sensors/dbm/ironware.inc.php]: That's a lot of _replaces, what's the reason?

Comment (thecityofguanyu) [reply includes/discovery/sensors/dbm/ironware.inc.php]: To strip off unnecessary data from the string returned in the snmpwalk. The devices send optical you alarm state, "dBm, plenty of whitespace, and wraps it all in double quotes.

Example output:

```
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.1 = STRING: " -006.5580 dBm: Normal"
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.2 = STRING: " -214748.3647 dBm: Low-Alarm "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.3 = STRING: " -006.5777 dBm: Normal"
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.4 = STRING: "  000.6625 dBm: Normal"
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.5 = STRING: "     N/A     "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.6 = STRING: "     N/A     "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.7 = STRING: "     N/A     "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.8 = STRING: "     N/A     "
FOUNDRY-SN-SWITCH-GROUP-MIB::snIfOpticalMonitoringTxPower.9 = STRING: "     N/A     "
```

If you have a recommendation to consolidate those various replace functions into less statements, then let me know.

Comment (laf) [reply includes/discovery/sensors/dbm/ironware.inc.php]: Maybe we should put something like this:

https://github.com/librenms/librenms/pull/7384/files#diff-67aaf7dd786395ce4cbd2c3996cf1bdcR1101

In a function `return_num()`?

Comment (thecityofguanyu) [reply includes/discovery/sensors/dbm/ironware.inc.php]: This definitely isn't the first time I've needed to rip out non-numeric characters, so an added function for that sounds like a great idea.

Comment (thecityofguanyu) [reply includes/discovery/sensors/dbm/ironware.inc.php]: I can start writing something up and add it to my PR, no big deal. Just correct/improve it as you see fit.

Comment (laf) [reply includes/discovery/sensors/dbm/ironware.inc.php]: 👍

**Comment by laf** [comment]
I've made some tweaks. Can you pull those changes in and see if it still works for you?

**Comment by thecityofguanyu** [comment]
Looks like it still works. Appreciate the fixes!

**Pull request merged**